app/controllers/images_controller.rb

```diff
@@ -72,28 +72,38 @@ def show
 
   def update
     @warpable = Warpable.find params[:warpable_id]
-
-    nodes = []
-    author = @warpable.map.author
-
-    # is it really necessary to make new points each time?
-    params[:points].split(':').each do |point|
-      lon = point.split(',')[0]
-      lat = point.split(',')[1]
-      node = Node.new(color: 'black',
-                      lat: lat,
-                      lon: lon,
-                      author: author,
-                      name: '')
-      node.save
-      nodes << node
+    map = Map.find(@warpable.map_id)
+    if map.anonymous? || logged_in?
+      nodes = []
+      author = @warpable.map.author
+      # is it really necessary to make new points each time?
+      params[:points].split(':').each do |point|
+        lon = point.split(',')[0]
+        lat = point.split(',')[1]
+        node = Node.new(color: 'black',
+                        lat: lat,
+                        lon: lon,
+                        author: author,
+                        name: '')
+        node.save
+        nodes << node
+      end
+      @warpable.count_version = 0
```

Comment:
@divyabaid16 I gone through this PR and implementation seems neat. Great work, but I wasn't able to figure out the use of count_version. Please clarify.

Reply:
@gauravano 
So basically, this count_version is present in the update with a value zero. This is to make sure that the version is stored only when Updation is performed. Because currently, if I revert to a particular version, the X and the Y coordinates are changing and that is also considered as an update. So on reverting, one more cloned version is created. 

This was present just to specify that create a version whenever the value of ```version_count``` is zero. And for this to work, in ```revert``` function definition, I will have to add ```@warpable.count_version = 1``` so that it doesn't create any cloned copy and everything works in a smooth fashion.

So basically this is done just to prevent the extra cloning. Does it makes sense now?
Thanks!